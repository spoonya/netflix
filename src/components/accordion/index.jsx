import React, { createContext, useState, useContext } from 'react';
import {
  Container,
  Item,
  Body,
  Header,
  Inner,
  Title
} from './styles/accordion';

const ToggleCtx = createContext();

export default function Accordion(props) {
  const { children } = props;

  return (
    <Container>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function FaqTitle(props) {
  const { children } = props;

  return <Title>{children}</Title>;
};

Accordion.Item = function FaqItem(props) {
  const [toggle, setToggle] = useState(false);
  const { children } = props;

  return (
    <ToggleCtx.Provider value={{ toggle, setToggle }}>
      <Item>{children}</Item>
    </ToggleCtx.Provider>
  );
};

Accordion.Header = function FaqHeader(props) {
  const { children } = props;
  const { toggle, setToggle } = useContext(ToggleCtx);

  return (
    <Header onClick={() => setToggle((prev = toggle) => !prev)}>
      {children}
      {toggle ? (
        <img src="/images/icons/close.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function FaqBody(props) {
  const { children } = props;
  const { toggle } = useContext(ToggleCtx);

  return toggle ? <Body>{children}</Body> : null;
};
