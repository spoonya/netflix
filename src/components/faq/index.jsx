import React, { createContext, useState, useContext } from 'react';
import { Container, Item, Body, Header, Inner, Title } from './styles/faq';

const ToggleCtx = createContext();

export default function Faq(props) {
  const { children } = props;

  return (
    <Container>
      <Inner>{children}</Inner>
    </Container>
  );
}

Faq.Title = function FaqTitle(props) {
  const { children } = props;

  return <Title>{children}</Title>;
};

Faq.Item = function FaqItem(props) {
  const [toggle, setToggle] = useState(false);
  const { children } = props;

  return (
    <ToggleCtx.Provider value={{ toggle, setToggle }}>
      <Item>{children}</Item>
    </ToggleCtx.Provider>
  );
};

Faq.Header = function FaqHeader(props) {
  const { children } = props;
  const { toggle, setToggle } = useContext(ToggleCtx);

  return (
    <Header onClick={() => setToggle((prev = toggle) => !prev)}>
      {children}
      {toggle ? (
        <img src="/images/icons/close.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Faq.Body = function FaqBody(props) {
  const { children } = props;
  const { toggle } = useContext(ToggleCtx);

  return toggle ? <Body>{children}</Body> : null;
};
