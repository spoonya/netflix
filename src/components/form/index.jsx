import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit
} from './styles/form';

export default function Form(props) {
  const { children } = props;

  return <Container>{children}</Container>;
}

Form.Base = function FormBase(props) {
  const { children, onSubmit } = props;

  return <Base onSubmit={onSubmit}>{children}</Base>;
};

Form.Error = function FormError(props) {
  const { children } = props;

  return <Error>{children}</Error>;
};

Form.Title = function FormTitle(props) {
  const { children } = props;

  return <Title>{children}</Title>;
};

Form.Text = function FormText(props) {
  const { children } = props;

  return <Text>{children}</Text>;
};

Form.TextSmall = function FormTextSmall(props) {
  const { children } = props;

  return <TextSmall>{children}</TextSmall>;
};

Form.Link = function FormLink(props) {
  const { children, to } = props;

  return <Link to={to}>{children}</Link>;
};

Form.Input = function FormInput(props) {
  const { children, placeholder, type, autoComplete } = props;

  return (
    <Input placeholder={placeholder} type={type} autoComplete={autoComplete}>
      {children}
    </Input>
  );
};

Form.Submit = function FormSubmit(props) {
  const { children, disabled } = props;

  return <Submit disabled={disabled}>{children}</Submit>;
};
