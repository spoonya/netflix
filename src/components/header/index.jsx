import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Container,
  Background,
  Frame,
  Logo,
  ButtonLink
} from './styles/header';

export default function Header(props) {
  const { children, bg } = props;

  return bg ? children : <Background>{children}</Background>;
}

Header.Frame = function HeaderFrame(props) {
  const { children } = props;

  return <Container>{children}</Container>;
};

Header.Logo = function HeaderLogo(props) {
  const { to, src, alt } = props;

  return (
    <ReactRouterLink to={to}>
      <Logo src={src} alt={alt} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink(props) {
  const { children, to } = props;

  return <ButtonLink to={to}>{children}</ButtonLink>;
};
