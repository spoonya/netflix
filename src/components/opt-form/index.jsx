import React from 'react';
import { Container, Input, Button, Text } from './styles/opt-form';

export default function OptForm(props) {
  const { children } = props;

  return <Container>{children}</Container>;
}

OptForm.Input = function OptFormInput() {
  return <Input />;
};

OptForm.Button = function OptFormButton(props) {
  const { children } = props;

  return (
    <Button>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try now" />
    </Button>
  );
};

OptForm.Text = function OptFormText(props) {
  const { children } = props;

  return <Text>{children}</Text>;
};
