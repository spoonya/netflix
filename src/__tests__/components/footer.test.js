import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../components';

describe('<Footer />', () => {
  it('renders the <Footer />', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us.</Footer.Title>
        <Footer.Row>
          <Footer.Col>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Col>

          <Footer.Col>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Col>

          <Footer.Col>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">Corporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Col>

          <Footer.Col>
            <Footer.Link href="#">Media Center</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Col>
        </Footer.Row>
        <Footer.Text>Netflix Belarus</Footer.Text>
      </Footer>,
    );

    expect(getByText('Questions? Contact us.')).toBeTruthy();

    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Investor Relations')).toBeTruthy();
    expect(getByText('Privacy')).toBeTruthy();
    expect(getByText('Speed Test')).toBeTruthy();

    expect(getByText('Help Center')).toBeTruthy();
    expect(getByText('Jobs')).toBeTruthy();
    expect(getByText('Cookie Preferences')).toBeTruthy();
    expect(getByText('Legal Notices')).toBeTruthy();

    expect(getByText('Account')).toBeTruthy();
    expect(getByText('Ways to Watch')).toBeTruthy();
    expect(getByText('Corporate Information')).toBeTruthy();
    expect(getByText('Netflix Originals')).toBeTruthy();

    expect(getByText('Media Center')).toBeTruthy();
    expect(getByText('Terms of Use')).toBeTruthy();
    expect(getByText('Contact Us')).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
