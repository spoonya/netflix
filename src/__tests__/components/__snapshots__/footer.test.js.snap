// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> 1`] = `
<footer
  class="footer__Container-qu6i6-0 bLHXco"
>
  <h2
    class="footer__Title-qu6i6-3 hcXouZ"
  >
    Questions? Contact us.
  </h2>
  <div
    class="footer__Row-qu6i6-2 hQISfg"
  >
    <div
      class="footer__Col-qu6i6-1 gJGTZT"
    >
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Col-qu6i6-1 gJGTZT"
    >
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Help Center
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Legal Notices
      </a>
    </div>
    <div
      class="footer__Col-qu6i6-1 gJGTZT"
    >
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Col-qu6i6-1 gJGTZT"
    >
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Media Center
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-qu6i6-5 jeTrZb"
        href="#"
      >
        Contact Us
      </a>
    </div>
  </div>
  <p
    class="footer__Text-qu6i6-4 agZJJ"
  >
    Netflix Belarus
  </p>
</footer>
`;
